### PR TITLE
[LLVM] add 12.0.1

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Comment by vchuravy** [comment]
needs https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/164

**Comment by giordano** [comment]
> needs JuliaPackaging/BinaryBuilderBase.jl#164

Why? The only failure I see looks unrelated

**Comment by vchuravy** [comment]
Oops it is of course https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/157

**Comment by giordano** [comment]
You should insist there, Elliot didn't look too much convinced 😛

**Comment by giordano** [comment]
Can you checkout `https://github.com/giordano/BinaryBuilderBase.jl.git#mg/csl_paths` in this branch so that we can see if https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/157 breaks anything?  If it doesn't that could be an argument in favour of the change :slightly_smiling_face:  (Yes, I'm suggesting testing in production :grin:)

**Comment by giordano** [comment]
```console
sandbox:${WORKSPACE}/build # echo $LD_LIBRARY_PATH 
/usr/lib/csl-musl-x86:/usr/lib/csl-glibc-x86:/usr/lib/csl-musl-i686_64:/usr/lib/csl-glibc-i686_64:/usr/local/lib64:/usr/local/lib:/usr/lib64:/usr/lib:/lib64:/lib:/workspace/x86_64-linux-musl-cxx11/destdir/lib:/opt/x86_64-linux-musl/x86_64-linux-musl/lib64:/opt/x86_64-linux-musl/x86_64-linux-musl/lib:/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/lib:/opt/aarch64-apple-darwin20/lib:/workspace/destdir/lib64:/workspace/destdir/lib
sandbox:${WORKSPACE}/build # ls /usr/lib/csl-musl-x86
ls: /usr/lib/csl-musl-x86: No such file or directory
sandbox:${WORKSPACE}/build # ls /usr/lib/csl-glibc-x86
ls: /usr/lib/csl-glibc-x86: No such file or directory
sandbox:${WORKSPACE}/build # ls /usr/lib/csl-musl-i686_64
ls: /usr/lib/csl-musl-i686_64: No such file or directory
sandbox:${WORKSPACE}/build # ls /usr/lib/csl-glibc-i686_64
ls: /usr/lib/csl-glibc-i686_64: No such file or directory
```
Well, I guess we need first to fix the CSL paths to point to something existing...

**Comment by giordano** [comment]
`aarch64-apple-darwin` is still failing :disappointed:

File: .ci/Manifest.toml [hunk 1]
```diff
@@ -36,11 +36,11 @@ version = "0.4.0"
 
 [[BinaryBuilderBase]]
 deps = ["CodecZlib", "Downloads", "InteractiveUtils", "JSON", "LibGit2", "Libdl", "Logging", "OutputCollectors", "Pkg", "Random", "SHA", "Scratch", "SimpleBufferStream", "TOML", "Tar", "UUIDs", "p7zip_jll", "pigz_jll"]
-git-tree-sha1 = "225b107285e25a745047582d138cd99986f52f1a"
```
Comment (giordano) [inline .ci/Manifest.toml:39]: Next time you update this pull request you can revert this change, as it is now live here

**Comment by vchuravy** [comment]
cc: @femtomc for MLIR

**Comment by vchuravy** [comment]
@Gnimuc does this look like it has the things you need?
